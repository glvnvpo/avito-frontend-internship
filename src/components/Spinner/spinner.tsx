// @flow

import React, {FC, HTMLAttributes} from 'react';
import {Spin} from 'antd';
import './styles.scss';

export enum Sizes {
    LARGE= 'large',
    SMALL = 'small'
}

export enum Colors {
    ORANGE= 'orange',
    GREY = 'grey',
    BLUE = 'blue'
}

type Props = {
    size?: Sizes;
	color?: Colors;
}

export const Spinner: FC<Props & HTMLAttributes<any>> = ({size = Sizes.LARGE, color = Colors.ORANGE,
															 ...rest}) => {

	return (
		<div className={`spinner ${color}`}>
			<Spin size={size} {...rest} />
		</div>
	);
};