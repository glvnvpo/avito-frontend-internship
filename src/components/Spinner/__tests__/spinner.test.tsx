import React from 'react';
import {mount} from 'enzyme';
import {Spinner} from '../spinner';

describe('Spinner', () => {
	let wrapper;

	it('should render default Spinner', () => {
		wrapper = mount(<Spinner/>);
		expect(wrapper).toMatchSnapshot();
	});
});