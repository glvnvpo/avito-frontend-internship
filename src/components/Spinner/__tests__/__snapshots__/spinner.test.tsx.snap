// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render default Spinner 1`] = `
<Spinner>
  <div
    className="spinner orange"
  >
    <Spin
      size="large"
    >
      <Spin
        size="large"
        spinPrefixCls="ant-spin"
      >
        <div
          aria-busy={true}
          aria-live="polite"
          className="ant-spin ant-spin-lg ant-spin-spinning"
        >
          <span
            className="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              className="ant-spin-dot-item"
            />
            <i
              className="ant-spin-dot-item"
            />
            <i
              className="ant-spin-dot-item"
            />
            <i
              className="ant-spin-dot-item"
            />
          </span>
        </div>
      </Spin>
    </Spin>
  </div>
</Spinner>
`;
